Core/Object Arts/Dolphin/Base/FloatTest.cls

```diff
@@ -361,7 +361,18 @@ testPrintOnDecimalPlaces
 	"Defect #373 - produced 3"
 	stream := String writeStream.
 	3.999 printOn: stream decimalPlaces: 0.
-	self assert: stream contents equals: '4'!
+	self assert: stream contents equals: '4'.
+
+	"Previous implementation was failing when sending to zero."
+	stream := String writeStream.
+	0.0 printOn: stream decimalPlaces: 0.
+	self assert: stream contents equals: '0'.
+	stream := String writeStream.
+	0.0 printOn: stream decimalPlaces: 1.
+	self assert: stream contents equals: '0.0'.
+	stream := String writeStream.
+	0.0 printOn: stream decimalPlaces: 2.
```

Comment:
Thanks for updating the unit tests. 
At least one test case for -0.0 (negative zero) would be useful. I think your code will work, but it is a useful regression test.